### ISSUE TITLE
output container doesn't work if target dir doesn't exist

### ISSUE THREAD

**Issue by reinhold-willcox** [TeamCOMPAS/COMPAS#998]
**Describe the bug**
This is a small one. If you run compas with `-o new/output/directory`, but that directory doesn't already exist, COMPAS ignores the command and uses the default CWD. 

**Label the issue**
`urgency_low`      - This issue is not urgent
`severity_minor`    - This is a minor bug with minimal impact

**To Reproduce**
Run `COMPAS -o directory/that/doesnt/exist`

**Expected behavior**
COMPAS should try to create the directory (if possible), or print a statement saying that it can't (if there are user privilege issues). 

**Versioning (please complete the following information):**
 - COMPAS v02.39.01

**Comment by jeffriley** [comment]
Huh. Never noticed...  Shouldn't be too hard to fix.

**Comment by jeffriley** [comment]
@reinhold-willcox 

Actually... Now that I've looked at this I remember I wrote it this way deliberately - it's not described in the documentation (oops), but the way the code is we expect the output directory to exist, and if it doesn't we just use the CWD.  If we don't require the output directory to exist, the logging code has to create it before creating files in it - which it can do, but that could mean if there is an error later we've created a directory that could be empty, and I didn't want that to happen.  The workaround is to ~~`touch`~~ *make* the directory (with `mkdir`) before running `COMPAS`, and `COMPAS` will run ok.  But if that's too unwieldy, I could change the code so that the logging code creates the directory before using it - it just may be that if something goes wrong (EDIT: or there are no stars/binaries to be written to the logfiles) there may be a circumstance where `COMPAS` leaves behind an empty directory.  Messy, but probably not the end of the world...

Thoughts?
